core/Rename.cpp

```diff
@@ -272,33 +268,52 @@ namespace olympia
 
                 const auto & srcs = renaming_inst->getSourceOpInfoList();
                 const auto & dests = renaming_inst->getDestOpInfoList();
-
                 for(const auto & src : srcs)
                 {
-                    const auto rf  = olympia::coreutils::determineRegisterFile(src);
-                    const auto num = src.field_value;
-                    auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    const uint32_t prf = map_table_[rf][num];
-                    reference_counter_[rf][prf]++;
-                    renaming_inst->getRenameData().setSource({prf, rf});
-                    bitmask.set(prf);
-
-                    ILOG("\tsetup source register bit mask "
-                         << sparta::printBitSet(bitmask)
-                         << " for '" << rf << "' scoreboard");
+                    if(renaming_inst->isStoreInst() && dests.size() == 0 && (src.field_id == mavis::InstMetaData::OperandFieldID::RS2)){
```

Comment:
Not sure why you're checking for dest list size here.  Stores will never have dests...

Reply:
I check the dest list because isStoreInst() is true for both loads and stores:
https://github.com/riscv-software-src/riscv-perf-model/blob/f18ab54d6c7a1b5d5b058e58577ee3141ca2533c/core/Inst.hpp#L173

so I check `dests.size() == 0` to make sure it's a store

Reply:
Fix in [e7fa84c](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/e7fa84ca02af05a0422db09893bfd2815f6b2e11)

I saw an increase in IPC (1.13 -> 1.4), just from fixing the logic behind isStoreInst, not sure if there was another bug induced by that fix, so might need additional checking.